### PR TITLE
UI Publish + Lockdown Published Fields

### PR DESCRIPTION
**Pull request by dheniges**
- Prevents deleting published fields
- Prevents updating published fields
- Plugs in UI publish button to publish endpoint
- Fixes an ordering bug with publish

### CODE REVIEW

**Comment by dheniges** [comment]
@jjidt ready for review whenever you get time today

File: api/src/businesstime/field.test-i.js [hunk 1]
```diff
@@ -150,6 +150,27 @@ describe('BusinessField', () => {
...
+          docId: updateFactoryDocument.id,
+          type: factoryField.type,
+          versionId: version.id
+        }))[0]
+      })
+
+      it('should throw an error', async () => {
+        await expect(
+          BusinessField.updateByIdForDocument(publishedField.id, docId, orgId, updateBody)
+        ).rejects.toThrow()
```
Comment (jjidt) [inline api/src/businesstime/field.test-i.js]: we should probably check the error code here

.rejects.toEqual(expect.objectContaining({ code: 403 }))

File: api/src/businesstime/field.test-i.js [hunk 2]
```diff
@@ -331,6 +352,28 @@ describe('BusinessField', () => {
...
+          docId: documentToDelete.id,
+          type: fieldC.type,
+          versionId: version.id
+        }))[0]
+      })
+
+      it('should throw an error', async () => {
+        await expect(
+          BusinessField.deleteByIdForDocument(publishedField.id, documentToDelete.id, constants.ORG_ID)
+        ).rejects.toThrow()
```
Comment (jjidt) [inline api/src/businesstime/field.test-i.js]: same

**Review by jjidt — CHANGES_REQUESTED** [review]
Fun to see this working, everything looking good, just a couple specific test value checks

**Review by jjidt — APPROVED** [review]
nice 👍

**Pull request merged**